include/nbl/video/IPhysicalDevice.h

```diff
@@ -581,33 +581,15 @@ class NBL_API2 IPhysicalDevice : public core::Interface, public core::Unmovable
             );
         }
 
-        template<typename FORMAT_USAGE>
-        struct FormatPromotionRequest
-        {
+        struct SBufferFormatPromotionRequest {
             asset::E_FORMAT originalFormat = asset::EF_UNKNOWN;
-            FORMAT_USAGE usages = FORMAT_USAGE();
```

Comment:
@Erfan-Ahmadi remind me why FORMAT_USAGE was a template parameter?

Reply:
pretty sure it was to have equal and hash defined.

Reply:
hmm ok, because if it wasn't template dependant then if would complain about the lack of definition for `std::hash`

Reply:
that's new behavior for me.